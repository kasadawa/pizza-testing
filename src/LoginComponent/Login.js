import  React from 'react';
import * as  $ from 'jquery';


class Login extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      'email':'',
      'password':''
    }
  }

  submitData(e){
    e.preventDefault(); // in order not to refresh the page  
    if (this.checkData()) this.makeReq(); 

  }
  checkData(){
    if(this.state.email !== '' && this.state.password !=='') return true ; 
    return false ;
    // here we can check the data if we want to
    // check email and password fields 
  }
  makeReq(){
    $.ajax({
      method:'POST',
      url : 'http://localhost:3001/login',
      data: this.state})
      .done((msg)=>{
        // if there is no error
        if(!!msg.error){
          // userLogin();
          alert(msg.data);
        }else{
          if(msg.data === 'success'){
            alert('successfuly loged')
          }
        }
      })
      .fail((res)=>{alert('something happend and the req failed' + res);})
  }

  changeHandler(e,stateString){
    this.setState({ [stateString]: e.target.value});
  }

  render() {
    return (
      <div className='row justify-content-md-center'>
        <div className="col-md-6">
        <form>
          <div className="form-group">
            <label htmlFor="exampleInputEmail1">Email address</label>
            <input value={this.state.email} onChange={(e)=>this.changeHandler(e,'email')} type="email" className="form-control" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email" />
          </div>
          <div className="form-group">
            <label htmlFor="exampleInputPassword1">Password</label>
            <input value={this.state.password} onChange={(e)=>this.changeHandler(e,'password')} type="password" className="form-control" id="exampleInputPassword1" placeholder="Password" />
          </div>
          <button id="submitData" type="submit" onClick={(e)=>this.submitData(e)} className="btn btn-primary">Submit</button>
        </form>
        </div>
      </div>
    );
  }
}

export default Login;



