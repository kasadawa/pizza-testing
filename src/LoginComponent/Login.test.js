// test  makeReq(){

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow,render, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


Enzyme.configure({adapter: new Adapter()});

import Login from './Login';


test('Login component snapshot', () => {
  const component = renderer.create(
    <Login></Login>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

describe('Testing Login component',()=>{

    
    it('Sending empty email and password,it should block me', () => {
        
       
        const wrapper = shallow(<Login />);
        const instance =  wrapper.instance(); 
        var creds = {email:'',password:''};
        wrapper.setState(creds);

        expect(instance.checkData()).toBe(false);

        creds = {email:'dkosadkaod',password:''};
        wrapper.setState(creds);
        expect(instance.checkData()).toBe(false);



        creds = {email:'',password:'dsadsako'};
        wrapper.setState(creds);
        expect(instance.checkData()).toBe(false);



        creds = {email:'kdosakdoa',password:'dsadsako'};
        wrapper.setState(creds);
        expect(instance.checkData()).toBe(true);
        

    });

});

 