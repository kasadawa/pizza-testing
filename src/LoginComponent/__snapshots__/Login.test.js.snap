// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component snapshot 1`] = `
<div
  className="row justify-content-md-center"
>
  <div
    className="col-md-6"
  >
    <form>
      <div
        className="form-group"
      >
        <label
          htmlFor="exampleInputEmail1"
        >
          Email address
        </label>
        <input
          aria-describedby="emailHelp"
          className="form-control"
          id="exampleInputEmail1"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="exampleInputPassword1"
        >
          Password
        </label>
        <input
          className="form-control"
          id="exampleInputPassword1"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        id="submitData"
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
